### PR TITLE
Fix connection placeholder replacements errors

### PR DESCRIPTION
**Pull request by edmocosta**
## Release notes

Fixed connection placeholder replacements errors when running Logstash 8.15.1 and 8.15.2 versions

## What does this PR do?

Logstash `8.15.1` changed the [replace_placeholders](https://github.com/elastic/logstash/blob/main/logstash-core/lib/logstash/util/substitution_variables.rb#L54) method adding a new boolean argument to it.
The plugins is  still [calling](https://github.com/logstash-plugins/logstash-input-azure_event_hubs/blob/main/lib/logstash/inputs/azure_event_hubs.rb#L335) that method with a single argument, which leads to an error and failure to initialize Logstash.

## Checklist

- [x] My code follows the style guidelines of this project
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] I have made corresponding change to the default configuration files (and/or docker env variables)
- [ ] I have added tests that prove my fix is effective or that my feature works

## How to test this PR locally

 - Run a simple pipeline using this plugins with Logstash 8.15.1 and previous version. 

## Related issues

- Closes https://github.com/logstash-plugins/logstash-input-azure_event_hubs/issues/91

## Logs

```
Error parsing event hub string name for connection: '' please ensure that the connection string contains the EntityPath
```

### CODE REVIEW

File: lib/logstash/inputs/azure_event_hubs.rb [hunk 1]
```diff
@@ -361,6 +360,24 @@ def initialize(params)
...
   attr_reader :event_hubs_exploded
 
+  def replace_connection_placeholders(value)
+    connection = value
+    if self.class.respond_to? :replace_placeholders
+      # Logstash 6.x
+      if self.class.method(:replace_placeholders).arity == 1
+        connection = self.class.replace_placeholders(connection)
+      else
+        # Logstash >= 8.15.1 changed the method arity including a new boolean parameter `refine`.
```
Comment (jsvd) [inline lib/logstash/inputs/azure_event_hubs.rb:370]:
```suggestion
        # Logstash 8.15.1 and 8.15.2 changed the method arity including a new boolean parameter `refine`. This was fixed in 8.15.3. see TODO PR TO CORE FIX
```

**Review by jsvd — APPROVED** [review]
LGTM, I recommend only merging and publishing once https://github.com/elastic/logstash/pull/16485 is merged and backported.

**Comment by JulienOrain** [comment]
Hello
I am not familiar with the release process of logstash and logstash plugins.
Do you know when this MR and the version `v1.4.8` will be release in logstash (probably in a `8.15.3`) ?
Thanks

**Comment by JulienOrain** [comment]
Awesome ! Thanks a lot !

**Pull request merged**